Description: Use exact compiler for dkms as used to build the kernel
 Ubuntu kernel builds do not export .kernelvariables file, thus extend
 CC detection to read compiler binary from the .config file


Index: dkms-2.8.7/dkms
===================================================================
--- dkms-2.8.7.orig/dkms
+++ dkms-2.8.7/dkms
@@ -1331,6 +1331,13 @@ prepare_build()
             $"Check $dkms_tree/$module/$module_version/build/ for more information."
     done
 
+    if [[ -e $kernel_source_dir/.config ]]; then
+        cc=$(sed -n 's|^CONFIG_CC_VERSION_TEXT="\([^ ]*\) .*"|\1|p' $kernel_source_dir/.config)
+        if command -v "$cc" >/dev/null; then
+            export CC="$cc"
+        fi
+    fi
+
     # Run the pre_build script
     run_build_script pre_build "$pre_build"
 }
